src/main/scala/co/teapot/tempest/server/TempestDatabaseClient.scala

```diff
@@ -17,64 +17,116 @@ package co.teapot.tempest.server
 import java.sql.{Connection, Types}
 
 import anorm._
-import co.teapot.tempest.{InvalidArgumentException, SQLException, UndefinedAttributeException}
+import co.teapot.tempest._
+import co.teapot.tempest.typedgraph.IntNode
 import org.postgresql.util.PSQLException
 
 import scala.collection.mutable
 
+/**
+  * A TempestDatabaseClient providems methods to query attributes of nodes, find nodes matching a given query, and
+  * convert between thrift Node objects (which have a string id chosen by the user) and internal IntNode opbjects
+  * (which have an Int tempestId assigned by tempest). */
+
 trait TempestDatabaseClient {
-  /** Returns a map from node id to attribute value for the the given attribute name.  Any nodes
-    * with null attribute value will be omitted from the resulting map.
-    * The implicit anorm.Column[A] parameter just means that A must be some type which Anorm
-    * can extract from a JDBC row. */
-  def getMultiNodeAttribute[A](graphName: String, nodeIds: Seq[Long], attributeName: String)
-                              (implicit c : anorm.Column[A]): collection.Map[Long, A]  // TODO: Delete method
-
-  def getMultiNodeAttributeAsJSON(graphName: String, nodeIds: Seq[Long], attributeName: String): collection.Map[Long, String]
-
-  def getNodeAttribute[A](graphName: String, nodeId: Long, attributeName: String)
-                         (implicit c : anorm.Column[A]): Option[A] =
-    getMultiNodeAttribute(graphName, Seq(nodeId), attributeName).get(nodeId)
-
-  def getNodeAttributeAsJSON(graphName: String, nodeId: Long, attributeName: String): String =
-    getMultiNodeAttributeAsJSON(graphName, Seq(nodeId), attributeName).getOrElse(nodeId, "null")
-
-  /*  If we create a mock database, we might need to go back to these methods:
-    def getNodeStringAttribute(nodeId: Long, attributeName: String): String =
-    getNodeAttribute(nodeId, attributeName, SqlParser.str(1))
-  def getNodeLongAttribute(nodeId: Long, attributeName: String): Long =
-    getNodeAttribute(nodeId, attributeName, SqlParser.long(1))
-  def getNodeBooleanAttribute(nodeId: Long, attributeName: String): Boolean =
-    getNodeAttribute(nodeId, attributeName, SqlParser.bool(1))
-   */
-  def setNodeAttribute(graphName: String,
-                       nodeId: Long,
+
+  def nodeToIntNode(node: Node): IntNode = {
+    nodeToIntNodeOption(node).getOrElse(
+      throw new InvalidNodeIdException(s"No node in ${node.`type`} has id ${node.id}")
+    )
+  }
+
+  def nodeToTempestId(node: Node): Int =
+    nodeToIntNode(node).tempestId
+
+  def nodeToIntNodeOption(node: Node): Option[IntNode] = {
```

Comment:
Good call, that is more clear.  (I think I was prematurely optimizing, to avoid an Int <> JSON string conversion, but that isn't the bottleneck here.)